users/src/main/java/io/crate/operation/user/PrivilegeValidator.java

```diff
@@ -0,0 +1,434 @@
+            }
+            return true;
+        }
+
+        @Override
+        protected Boolean visitDropUserStatement(DropUserAnalyzedStatement analysis, PrivilegeContext context) {
+            if (!context.isSuperUser(context.sessionContext().user())) {
+                context.throwUnauthorized(context.sessionContext().user());
+            }
+            return true;
+        }
+
+        @Override
+        public Boolean visitPrivilegesStatement(PrivilegesAnalyzedStatement analysis, PrivilegeContext context) {
+            if (!context.isSuperUser(context.sessionContext().user())) {
+                context.throwUnauthorized(context.sessionContext().user());
+            }
+            return true;
+        }
+
+        @Override
+        public Boolean visitAlterTableStatement(AlterTableAnalyzedStatement analysis, PrivilegeContext context) {
+            context.userManager().raiseMissingPrivilegeException(Privilege.Clazz.TABLE,
+                Privilege.Type.DDL,
+                analysis.table().ident().toString(),
+                context.sessionContext().user());
+            return true;
+        }
+
+        @Override
+        protected Boolean visitCopyFromStatement(CopyFromAnalyzedStatement analysis, PrivilegeContext context) {
+            context.userManager().raiseMissingPrivilegeException(Privilege.Clazz.TABLE,
+                Privilege.Type.DML,
+                analysis.table().ident().toString(),
+                context.sessionContext().user());
+            return true;
+        }
+
+        @Override
+        protected Boolean visitCopyToStatement(CopyToAnalyzedStatement analysis, PrivilegeContext context) {
+            context.userManager().raiseMissingPrivilegeException(Privilege.Clazz.TABLE,
+                Privilege.Type.DQL,
+                analysis.subQueryRelation().tableRelation().getQualifiedName().toString(),
+                context.sessionContext().user());
+            return true;
+        }
+
+        @Override
+        protected Boolean visitCreateTableStatement(CreateTableAnalyzedStatement analysis, PrivilegeContext context) {
+            context.userManager().raiseMissingPrivilegeException(Privilege.Clazz.TABLE,
+                Privilege.Type.DDL,
+                analysis.tableIdent().toString(),
+                context.sessionContext().user());
+            return true;
+        }
+
+        @Override
+        protected Boolean visitCreateRepositoryAnalyzedStatement(CreateRepositoryAnalyzedStatement analysis, PrivilegeContext context) {
+            context.userManager().raiseMissingPrivilegeException(Privilege.Clazz.CLUSTER,
+                Privilege.Type.DDL,
+                null,
+                context.sessionContext().user());
+            return true;
+        }
+
+        @Override
+        protected Boolean visitDeleteStatement(DeleteAnalyzedStatement analysis, PrivilegeContext context) {
+            context.userManager().raiseMissingPrivilegeException(Privilege.Clazz.TABLE,
```

Comment:
@seut Delete Statement needs DML privilege and relationVisitor checks DQL privilege always. I  i will keep it for this case and call the relation visitor as well